src/main/java/vectorwing/farmersdelight/common/block/MushroomColonyBlock.java

```diff
@@ -34,16 +36,16 @@
 import net.neoforged.neoforge.common.Tags;
 import vectorwing.farmersdelight.common.tag.ModTags;
 
-@SuppressWarnings("deprecation")
 public class MushroomColonyBlock extends BushBlock implements BonemealableBlock
 {
 	public static final MapCodec<MushroomColonyBlock> CODEC = RecordCodecBuilder.mapCodec(
-			builder -> builder.group(BuiltInRegistries.ITEM.holderByNameCodec().fieldOf("mushroom").forGetter(block -> block.mushroomType), propertiesCodec())
+			builder -> builder.group(BuiltInRegistries.ITEM.byNameCodec().fieldOf("mushroom").forGetter(block -> block.mushroomType), propertiesCodec())
 					.apply(builder, MushroomColonyBlock::new)
 	);
 
+	public static HashMap<Item, MushroomColonyBlock> COLONIES = new HashMap<>();
 	public static final int PLACING_LIGHT_LEVEL = 13;
-	public final Holder<Item> mushroomType;
+	public final Item mushroomType;
```

Comment:
This is not a proper change, `Holder` should be used as addons could register mushroom colonies for their own mushrooms, where the item is registered after the blocks and thus not available in the block's constructor.

Also, the `COLONIES` field should not be exposed, use proper `register` and `get` with checks for duplicate registers, also since the `Item` is not available in the constructor, do not register to the `COLONIES` in the constructor, move it `FMLCommonSetupEvent` instead.

Since we are moving away from hardcoding, let's not restrict the value class to `MushroomColonyBlock`, and the key class could just use `Block` instead of `Item` since we are not doing the register in `MushroomColonyBlock`'s constructor anyways.